src/plugin/pid/pid_filewrappers.cpp

```diff
@@ -63,6 +63,10 @@ using namespace dmtcp;
 // FIXME:  This function needs third argument newpathsize, or assume PATH_MAX
 static void updateProcPathVirtualToReal(const char *path, char **newpath)
 {
+  if (! Util::isReadableMemoryAddress(path)) {
+    *newpath = const_cast<char *>(path);
+    return;
+  }
   if (Util::strStartsWith(path, PROC_PREFIX)) {
```

Comment:
I'm worried about keeping the intention of the code clear.  Right now, it's
```if (! Util::isReadableMemoryAddress(path)) { ...}
    else if (CaseA) {...}
    else {...}```
The logic of how to add CaseB is clear.  But if we write:
```if (CaseA && ! Util::isReadableMemoryAddress(path)) {...}
    else {...}```
will it be obvious for a naive programmer that
``` ! Util::isReadableMemoryAddress(path)```
must be added to every new case?  (I'm being a little paranoid, because much of the DMTCP code will continue to be written by new students, and so I want to keep the code "student-safe".  :-)